README.md

```diff
@@ -1,8 +1,40 @@
-# Simple Echo Webserver using the Boost API
+# Nginx Clone Webserver
+A webserver written in C++ using the Boost API. This webserver supports a common
+RequestHandler interface/abstract base class that can be used to easily implement
+any handler.
+
+**Authors**:
+* Ryan Peterman
+* Aditya Raju
+* Yilei Zhang
+
+## Important Commands
+To compile:
+
+``` make ```
+
+To run:
+
+``` ./web-server <config-file> ```
+
+To test:
+
+``` make test ```
+
+To get boost (Ubuntu):
 
-## To get Boost
 ``` sudo apt-get install libboost-all-dev ```
 
-## To compile & run
-``` make && ./server ```
+## Config File Format
+``` bash
+# port the webserver is listening on
+port <port-num>;
+
+# how to connected a url to a handler
+path <url-mapping> <handler-name> {
+    <additional parameters as expected by handler>
+}
 
+# Handler to be called if no url matches
+default <handler-name> {}
+```
```

Comment:
I think we should add info for what the server already supports (static handling, echo handling, status handling, 404 handling. That is something we can add at the end though when we finish anything.